src/content/docs/en/guides/content-collections.mdx

```diff
@@ -337,6 +337,31 @@ relatedPosts:
 ---
 ```
 
+### Defining custom IDs
+
+When using the `glob()` loader with Markdown, MDX, Markdoc, or JSON files, every content entry [`id`](/en/reference/modules/astro-content/#id) is automatically generated in an URL-friendly format based on the content filename.
+The `id` is used to query the entry directly from your collection.
+It is also useful when creating new pages and URLs from your content.
+
+You can override an entry’s generated `id` by adding your own `slug` property to the file frontmatter or data object for JSON files.
+This is similar to the “permalink” feature of other web frameworks.
+
+```md {3}
+---
+title: My Blog Post
+slug: my-custom-id/supports/slashes
+---
+Your blog post content here.
+```
+
+```json {3}
+{
+  "title": "Blog Category",
+  "slug": "my-custom-id/supports/slashes",
+  "description": "Your category description here."
+}
+```
```

Comment:
``````suggestion
When using the `glob()` loader with Markdown, MDX, Markdoc, or JSON files, every content entry [`id`](/en/reference/modules/astro-content/#id) is automatically generated in an URL-friendly format based on the content filename. The `id` is used to query the entry directly from your collection. It is also useful when creating new pages and URLs from your content.

You can override an entry’s generated `id` by adding your own `slug` property to the file frontmatter or data object for JSON files. This is similar to the “permalink” feature of other web frameworks.

```md title="src/blog/1.md" {3}
---
title: My Blog Post
slug: my-custom-id/supports/slashes
---
Your blog post content here.
```

```json title="src/categories/1.json" {3}
{
  "title": "My Category",
  "slug": "my-custom-id/supports/slashes",
  "description": "Your category description here."
}
```
``````
Added file titles, and combined sentences into paragraphs since the rest of docs is that way.